pkg/osv/osv.go

```diff
@@ -185,23 +187,74 @@ func Get(id string) (*models.Vulnerability, error) {
 	return &vuln, nil
 }
 
+type hydrateResponse struct {
+	batchIdx  int
+	resultIdx int
+	vuln      *models.Vulnerability
+	err       error
+}
+
 // Hydrate fills the results of the batched response with the full
 // Vulnerability details.
 func Hydrate(resp *BatchedResponse) (*HydratedBatchedResponse, error) {
-	// TODO(ochang): Parallelize requests, or implement batch GET.
 	hydrated := HydratedBatchedResponse{}
 
+	responseChan := make(chan hydrateResponse, maxConcurrentRequests)
+	rateLimiter := make(chan struct{}, maxConcurrentRequests)
+
+	// Run in separate goroutine so that we can immediately start
+	// listening to the responseChan, and it will be not blocked
+	go func(resp *BatchedResponse) {
+		for batchIdx, response := range resp.Results {
+			for resultIdx, vuln := range response.Vulns {
+				rateLimiter <- struct{}{}
```

Comment:
nit: can we use https://pkg.go.dev/golang.org/x/sync/semaphore#example-package-WorkerPool instead?

Reply:
I did see that package, but since semaphores are under the /x/ experimental package I'm not sure whether we should use it.

Reply:
Should be fair game to use it. There's a lot of prominent packages using it already: https://deps.dev/go/golang.org%2Fx%2Fsync/v0.1.0/dependents